src/dict.c

```diff
@@ -1218,6 +1218,112 @@ char *stringFromLongLong(long long value) {
     return s;
 }
 
+/* Evolving dict into dictX. */
+dictX *dictEvolution(dict *d) {
```

Comment:
maybe assert that `d` isn't being rehashed?

Reply:
For this test, I'm pretty sure the rehash is done, so I've omitted this assertion, which is incorrect, and I'll add.